File: src/app/Class/Localization.ts
export class Localization {
  private id: number;
  private country: string;
  private city: string;
  private address: string;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.country = (obj && obj.country) || "";
    this.city = (obj && obj.city) || "";
    this.address = (obj && obj.address) || "";
  }

  // Getters

  getId(): number {
    return this.id;
  }

  getCountry(): string {
    return this.country;
  }

  getCity(): string {
    return this.city;
  }

  getAddress(): string {
    return this.address;
  }

  // Setters

  setCountry(country: string) {
    this.country = country;
  }

  setCity(city: string) {
    this.city = city;
  }

  setAddress(address: string) {
    this.address = address;
  }
}

File: src/main/java/MusicCube/entities/Localization.java
package MusicCube.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;

@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Entity
public class Localization {

    @Id
    @GeneratedValue
    @Column
    private int id;

    // TODO Zmienienie bazy danych i tego entity by było połączone z entity country, też uważam, że ta klasa powinna już tylko konkretne miejscówki naprzykład areny muzyczne
    // TODO więc powinniśmy też zrobić odzielne entity dla miast, a tu dodać nazwę miejscówy.

    @Column
    private String country;

    @Column
    private String city;

    @Column
    private String address;

    public Localization() {}

    public Localization(String country, String city) {
        this.country = country;
        this.city = city;
    }

    public Localization(String country, String city, String address) {
        this.country = country;
        this.city = city;
        this.address = address;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }
}
